enhance(ResolveFieldValue): note that list items may be async (#1066)
graphql/graphql-spec: spec/Section 6 -- Execution.md

{"soulMate"}, and the {objectValue} representing John Lennon. It would be
expected to yield the value representing Yoko Ono.

> List values are resolved similarly. For example, {ResolveFieldValue} might also

Reviewer: I do not like "similarly" because technically it follows the exact same algorithm, whereas "similarly" implies that there _is_ a difference. Let's try alternate wording:

```suggestion
For an example relating to list fields, {ResolveFieldValue} might
```

Though I'm not convinced we need this paragraph.

Author: I just removed the whole paragraph.

Note: It is common for {resolver} to be asynchronous due to relying on reading
an underlying database or networked service to produce a value. This
necessitates the rest of a GraphQL executor to handle an asynchronous execution

> flow. In addition, an implementation for collections may leverage asynchronous
> iterators or asynchronous generators provided by many programming languages.

Reviewer: This would introduce the term "generator" and "iterator" to the spec, which isn't currently used. Let's keep the language simpler:

```suggestion
flow. In addition, for fields that have a return type that is a List type, each
value in a collection of values returned by {resolver} may itself be
asynchronous.
```

Author: What about something very similar:

```suggestion
flow. In addition, for fields that have a return type that is a List type, each
value in a collection of values returned by {resolver} may be
retrieved asynchronously.
```

This just avoids saying that the value itself is asynchronous, which I think you might mean to just be a more concise version of the above?

@robrichard may have input, this was extracted from #742

Note: It is common for {resolver} to be asynchronous due to relying on reading
an underlying database or networked service to produce a value. This
necessitates the rest of a GraphQL executor to handle an asynchronous execution

> flow. In addition, for fields that have a return type that is a List type, each
> value in a collection of values returned by {resolver} may be retrieved
> asynchronously.

Reviewer: Alternative wording suggestion:

```suggestion
flow. If the field is of a list type, each
value in the collection of values returned by {resolver} may itself be retrieved
asynchronously.
```

I don't like mixing a non-normative note with "may" (which is usually normative); but I think this is okay.

Author: Sounds good!
